pynxtools/dataconverter/helpers.py

```diff
@@ -424,17 +426,25 @@ def does_group_exist(path_to_group, data):
     return False
 
 
-def ensure_all_required_fields_exist(template, data):
+def ensure_all_required_fields_exist(template, data, nxdl_root):
     """Checks whether all the required fields are in the returned data object."""
     for path in template["required"]:
         entry_name = get_name_from_data_dict_entry(path[path.rindex('/') + 1:])
         if entry_name == "@units":
             continue
         nxdl_path = convert_data_converter_dict_to_nxdl_path(path)
         is_path_in_data_dict, renamed_path = path_in_data_dict(nxdl_path, data)
-        if path in template["lone_groups"] and does_group_exist(path, data):
-            continue
 
+        renamed_path = path if renamed_path is None else renamed_path
+        if path in template["lone_groups"]:
+            opt_parent = check_for_optional_parent(path, nxdl_root)
+            if not does_group_exist(renamed_path, data):
+                raise ValueError(f"The required group, {path}, hasn't been supplied.")
+            if opt_parent != "<<NOT_FOUND>>" and does_group_exist(opt_parent, data):
+                if not does_group_exist(renamed_path, data):
```

Comment:
Will this not always be catched by the first if-clause already?

Reply:
Yes. Thanks. I fixed it now. :pray: